src/ffetch.js

```diff
@@ -9,4 +8,128 @@
  * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
- */
\ No newline at end of file
+ */
+
+async function* request(url, context) {
+  const { chunks, fetch } = context;
+  for (let offset = 0, total = Infinity; offset < total; offset += chunks) {
+    const resp = await fetch(`${url}?offset=${offset}&limit=${chunks}`);
+    if (resp.ok) {
+      const json = await resp.json();
+      total = json.total;
+      for (const entry of json.data) yield entry;
+    } else {
+      return;
+    }
+  }
+}
+
+// Operations
+
+function chunks(upstream, context, chunks) {
+  context.chunks = chunks;
+  return upstream;
+}
+
+async function* skip(upstream, skip) {
+  let skipped = 0;
+  for await (const entry of upstream) {
+    if (skipped < skip) {
+      skipped += 1;
+    } else {
+      yield entry;
+    }
+  }
+}
+
+async function* limit(upstream, limit) {
+  let yielded = 0;
+  for await (const entry of upstream) {
+    yield entry;
+    yielded += 1;
+    if (yielded === limit) {
+      return;
+    }
+  }
+}
+
+async function* map(upstream, fn) {
+  for await (let entry of upstream) {
+    entry = await fn(entry);
+    if (entry) yield entry;
+  }
+}
+
+function filter(upstream, fn) {
+  return map(upstream, (entry) => (fn(entry) ? entry : null));
+}
+
+function slice(upstream, from, to) {
+  return limit(skip(upstream, from), to - from);
+}
+
+function follow(upstream, name, context) {
```

Comment:
It's unclear to me what the use case of `follow` is?

Reply:
It will pre-fetch it, but it will also parse it, so you can use content from the referenced document in your overview page.